internal/templates/assisted-installer/template.go

```diff
@@ -141,7 +141,8 @@ spec:
   searchCollector:
     enabled: false`
 
-const NMStateConfig = `apiVersion: agent-install.openshift.io/v1beta1
+const NMStateConfig = `{{ if .SpecialVars.CurrentNode.NodeNetwork }}
```

Comment:
Do we have some kind of unit test for this or no?

Reply:
Created https://issues.redhat.com/browse/CNF-14484 for adding test coverage for default AI and IBI templates!